generator/generator_tests/altitude_test.cpp

```diff
@@ -0,0 +1,232 @@
+#include "testing/testing.hpp"
+
+#include "generator/altitude_generator.hpp"
+
+#include "generator/generator_tests_support/test_feature.hpp"
+#include "generator/generator_tests_support/test_mwm_builder.hpp"
+
+#include "routing/road_graph.hpp"
+#include "routing/routing_helpers.hpp"
+
+#include "indexer/altitude_loader.hpp"
+#include "indexer/classificator_loader.hpp"
+#include "indexer/index.hpp"
+#include "indexer/feature_processor.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "platform/country_file.hpp"
+#include "platform/platform.hpp"
+
+#include "platform/platform_tests_support/writable_dir_changer.hpp"
+
+#include "base/logging.hpp"
+#include "base/scope_guard.hpp"
+
+#include "std/string.hpp"
+
+using namespace feature;
+using namespace generator;
+using namespace platform;
+using namespace routing;
+using namespace tests_support;
+
+namespace
+{
+// These tests generate mwms with altitude sections and then check if altitudes
+// in the mwms are correct. The mwms are inited with different sets of features defined below.
+// There are several restrictions for these features:
+// * all of them are linear
+// * all coords of these features should be integer (it's necessary for easy implementation of MockAltitudeGetter)
+// * if accoding to one feature a point has some altitude, the point of another feature
+//   with the same coords has to have the same altitude
+// @TODO(bykoianko) Add ability to add to the tests not road features without altitude information.
+
+// Directory name for creating test mwm and temporary files.
+string const kTestDir = "altitude_generation_test";
+// Temporary mwm name for testing.
+string const kTestMwm = "test";
+
+struct Rounded3DPoint
+{
+  Rounded3DPoint(int32_t x, int32_t y, TAltitude a) : point2D(x, y), altitude(a) {}
+
+  m2::PointI point2D;
```

Comment:
> s/point2D/point/g

Done

> Also, consider to add prefix "m_" to both fields.

Do you think it's worth using prefix "m_" in case when stuctures(classes) are used for grouping, but not for incapsulation? What's your opinion?